src/lib/slice.zig

```diff
@@ -420,6 +534,57 @@ pub const Slice = struct {
     }
 };
 
+// Helper functions for slice parent notes
+
+/// Parses slice parent note content and extracts parent branch name
+fn parseSliceParentNote(note_content: []const u8, alloc: Allocator) !?[]const u8 {
+    // Look for "slice-parent: " prefix
+    if (!std.mem.startsWith(u8, note_content, SLICE_PARENT_NOTE_PREFIX)) {
+        return null; // Not a valid slice parent note
+    }
+    
+    const parent_name = note_content[SLICE_PARENT_NOTE_PREFIX.len..];
+    
+    // Trim whitespace and validate
+    const trimmed = std.mem.trim(u8, parent_name, " \t\n\r");
+    if (trimmed.len == 0) {
+        return null; // Empty parent name
+    }
+    
+    return try alloc.dupe(u8, trimmed);
+}
+
+/// Finds all slices that have parent notes in the repository
+pub fn findSlicesWithParentNotes(repo: GitRepository, alloc: Allocator) ![]LibGit.GitOID {
+    var iterator = LibGit.GitNoteIterator.init(repo, null) catch |err| {
+        log.err("Failed to create note iterator: {}", .{err});
+        return err;
+    };
+    defer iterator.free();
+    
+    var slice_commits = std.ArrayList(LibGit.GitOID).init(alloc);
+    errdefer slice_commits.deinit();
+    
+    while (iterator.next() catch |err| {
+        log.err("Failed to iterate notes: {}", .{err});
+        return err;
+    }) |note_info| {
+        // Read the note to check if it's a slice parent note
+        if (LibGit.GitNote.read(repo, note_info.annotated_id) catch null) |note| {
```

Comment:
Instead of using `LibGit.GitNote` import `GitNote` below in the file and use it.